### PR TITLE
RequireJS migration: create/edit export

### PR DESCRIPTION
**Pull request by orangejenny**
This is branched off of https://github.com/dimagi/commcare-hq/pull/22603, first new commit is 2a2f6c2bcae566ec81084ef58489f9a9a69a46d9

@millerdev / @dannyroberts

### CODE REVIEW

File: corehq/apps/export/static/export/js/customize_export_new.js [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+    'export/js/models',
+], function (
+    $,
+    initialPageData,
+    models
+) {
     $(function () {
-        var ExportInstance = hqImport('export/js/models').ExportInstance;
-        var customExportView = new ExportInstance(
+        var customExportView = new models.ExportInstance(
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/export/static/export/js/customize_export_new.js]: nope, not right now

File: corehq/apps/hqwebapp/static/hqwebapp/js/validators.ko.js [hunk 2]
```diff
@@ -1,65 +1,71 @@
...
+    'knockout',
+    'knockout-validation/dist/knockout.validation.min', // needed for ko.validation
+], function (
+    $,
+    ko
+) {
+    ko.validation.rules['emailRFC2822'] = {
+        validator: function (val) {
+            if (val === undefined || val.length === 0) return true;  // do separate validation for required
+            var re = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])$/;
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/js/validators.ko.js]: ignoring this one, too

**Comment by orangejenny** [comment]
Rebased now that https://github.com/dimagi/commcare-hq/pull/22603 is merged. Leaving "don't merge" label on until that PR gets deployed.

File: corehq/apps/export/static/export/js/models.js [hunk 3]
```diff
@@ -26,6 +26,47 @@ hqDefine('export/js/models', [
     constants,
     utils
 ) {
+    /**
+     * readablePath
+     *
+     * Helper functio that takes an array of PathNodes and converts them to a string dot path.
```
Comment (millerdev) [inline corehq/apps/export/static/export/js/models.js]: nit: functio -> function

**Review by millerdev — APPROVED** [review]

File: corehq/apps/export/static/export/js/models.js (hunk 3 above)
Comment (orangejenny) [reply corehq/apps/export/static/export/js/models.js]: fixed

**Review by millerdev — APPROVED** [review]

**Pull request merged**